x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/process/MlMemoryTracker.java

```diff
@@ -101,6 +101,11 @@ private void setReassignmentRecheckInterval(TimeValue recheckInterval) {
     @Override
     public void onMaster() {
         isMaster = true;
+        try {
+            asyncRefresh();
+        } catch (Exception ex) {
```

Comment:
I don't think the `try`/`catch` here is really necessary.  The cluster applier service already wraps each callback to a listener in a `try`/`catch`, so an exception here wouldn't stop other listeners being called.  And `asyncRefresh` itself already catches the most likely exception it could throw.  So any exception caught here would be _really_ strange.  So I would just avoid the clutter and call `asyncRefresh()` outside a `try` block after the trace logging.